scripts/s2n_struct_mem.py

```diff
@@ -0,0 +1,94 @@
+#! /usr/bin/env python
+
+'''
+Script to examine the fields of a structure.
+Useful for identifying large fields in complex structures like s2n_connection.
+
+Example usage:
+./scripts/s2n_struct_mem.py -h : print the usage message
+./scripts/s2n_struct_mem.py s2n_connection --min 100 : list the size of all fields of s2n_connection larger than 100 bytes
```

Comment:
In the future it would be nice to get struct information from the whole codebase and then print that to a csv. It would make it easier to find unexpected results rather than knowing beforehand. We could also upload that csv somewhere and track struct size over time :)